core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraSystem.kt

```diff
@@ -911,7 +1329,7 @@ constructor(
     override fun getScreenFlashEvents() = screenFlashEvents
     override fun getCurrentSettings() = currentSettings.asStateFlow()
 
-    override fun setFlashMode(flashMode: FlashMode) {
+    override suspend fun setFlashMode(flashMode: FlashMode) {
```

Comment:
With this change, you're now calling a `suspend` function (`tryApplyDynamicRangeConstraints`) within a `MutableStateFlow.update{}` lambda, which internally uses a CAS loop. This could run multiple times depending on if the CameraAppSettings change during execution of the lambda, which is more likely if it takes longer, such as with a suspend function. Generally, you want to avoid any heavy computation within an `update` lambda.